code/modules/lighting/lighting_source.dm

```diff
@@ -6,10 +6,13 @@
 	var/atom/source_atom     // The atom that we belong to.
 
 	var/turf/source_turf     // The turf under the above.
-	var/light_power    // Intensity of the emitter light.
-	var/light_range      // The range of the emitted light.
-	var/light_color    // The colour of the light, string, decomposed by parse_light_color()
+	var/light_power    	// Intensity of the emitter light.
+	var/light_range     // The range of the emitted light.
+	var/light_color    	// The colour of the light, string, decomposed by parse_light_color()
 	var/light_uv		// The intensity of UV light, between 0 and 255.
+	var/light_angle		// The light's emission angle, in degrees.
+	var/light_dir = EAST		// The direction the light is facing.
```

Comment:
It doesn't get any information from the top atom yet, that was just for debugging.